portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,23 +34,57 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    List<String> favouriteFruits = Arrays.asList("Mango", "Strawberry", "Grapes");
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-       
-        //converts the arrayList to Json using convertToJsonUsingGson
-        String json = convertToJsonUsingGson(favouriteFruits);
+        Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+        
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<String> comments = new ArrayList<>();
+
+        for (Entity entity : results.asIterable()) {
+            String title = (String) entity.getProperty("title");
+            comments.add(title);
+        }
+
+        Gson gson = new Gson();
 
-        // Send the JSON as the response
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        response.getWriter().println(gson.toJson(comments));
     }
 
-    private String convertToJsonUsingGson(List<String> list) {
-        Gson gson = new Gson();
-        String json = gson.toJson(list);
-        return json;
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String text = getParameter(request, "comment", "");
```

Comment:
Add a check for if text = “”, since you probably don’t want to store empty strings in your data store :)